### ISSUE TITLE
Support MP3

### ISSUE THREAD

**Issue by pcwalton** [pcwalton/rust-media#10]
We should add MP3 support via FFmpeg and `AudioUnit.framework` on Mac. It may also be desirable to add ID3 support.

**Comment by cuzbog** [comment]
Is it ok if we don't use FFmpeg or AudioUnit.framework and write a decoder in Rust instead?

**Comment by pcwalton** [comment]
It is, but that decoder should be treated as a fallback for when no hardware decoder is available. Sadly, the only way to use HW decoders is typically the platform APIs, due to the closed proprietary nature of decoders :(

**Comment by cuzbog** [comment]
But isn't that only for Win + Mac? For Linux we're using software decoding anyways, aren't we?

**Comment by pcwalton** [comment]
Yeah, I don't think desktop Linux usually uses HW decoding, so a pure Rust solution sounds just fine there. Assuming it works and is fast enough, of course :)

**Comment by cuzbog** [comment]
It would be an interesting experiment :)

**Comment by cuzbog** [comment]
How would we go about per-frame headers and info (ex. sampling rate)? It is common practice to assume that every frame will be like the first, but that's not correct _per se_.

**Comment by cuzbog** [comment]
@pcwalton ^^

**Comment by pcwalton** [comment]
Usually I've been following the way mkv does these things. For example, the notion of cluster/track/etc. is taken from mkv. So I guess my answer would be "however mkv handles it".

**Comment by cuzbog** [comment]
@pcwalton Well, it seems like mkv handles it exactly the way I put it (at least judging by [this](http://www.matroska.org/news/compressed-headers.html)), which assumes that each frame has the same headers, but this is not correct, and it will not allow features such as variable bitrate

**Comment by ritiek** [comment]
Any update on MP3 support?

**Comment by jdm** [comment]
Nobody is working on rust-media at this point.

**Comment by lieff** [comment]
Hi, here new https://crates.io/crates/minimp3
